app/controllers/profiles_controller.rb

```diff
@@ -23,6 +23,15 @@ def search
     if params[:distance]
       @results = @results.near(current_customer.profile.postcode, params[:distance].to_f)
     end
+  
+    if params[:age]
+      @results = @results.select do |result| 
```

Comment:
Rather than using `select`, use a `where` and perform your searches at the DB level